core/src/main/scala/memeid/UUID.scala

```diff
@@ -55,59 +61,113 @@ sealed trait UUID extends Comparable[UUID] {
    * The variant number has the following meaning:
    *
    * - '''0''': Reserved for NCS backward compatibility
+   *
    * - '''2''': [[https://tools.ietf.org/html/rfc4122 IETF RFC 4122]]
+   *
    * - '''6''': Reserved, Microsoft Corporation backward compatibility
+   *
    * - '''7''': Reserved for future definition
    *
    * Interoperability, in any form, with variants other than the one
    * defined here is not guaranteed, and is not likely to be an issue in
    * practice.
    *
    * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.1]]
+   * @return The variant of this [[UUID]]
    */
   @inline def variant: Int = juuid.variant
 
+  /**
+   * The version number associated with this {@code UUID}.  The version
+   * number describes how this {@code UUID} was generated.
+   *
+   * The version number has the following meaning:
+   *
+   * - '''1''':    Time-based UUID
+   *
+   * - '''2''':     DCE security UUID
+   *
+   * - '''3''':    Name-based UUID
+   *
+   * - '''4''':   Randomly generated UUID
+   *
+   * - '''5''': The name-based version that uses SHA-1 hashing
+   *
+   * @see https://tools.ietf.org/html/rfc4122#section-4.1.3
+   * @return The version number of this [[UUID]]
+   */
   @inline def version: Int = juuid.version
 
+  /**
+   * Checks if an [[UUID]] passed as parameter is equal to this [[UUID]]
+   * @param obj object to be checked if it's equal to this [[UUID]]
+   * @return true if both are equals; false otherwise
+   */
   @SuppressWarnings(Array("scalafix:Disable.equals", "scalafix:Disable.Any"))
   override def equals(obj: Any): Boolean = obj match {
     case x: UUID if compareTo(x).equals(0) => true
     case _                                 => false
   }
 
+  /**
+   * Compares this [[UUID]] with another passed as parameter
+   * @param x [[UUID]] to be compared with this
+   * @return 0 if both are equals
+   *         a value less than 0 if this is lower than {@code x}
+   *         a value greater than 0 if this is higher than {@code x}
+   */
   override def compareTo(x: UUID): Int = {
     compareUnsigned(msb, x.msb) match {
       case 0     => compareUnsigned(lsb, x.lsb)
       case other => other
     }
   }
 
+  /**
+   * Returns a hash code for this [[UUID]]
+   *
+   * @return A hash code value for this [[UUID]]
+   */
   @SuppressWarnings(Array("scalafix:Disable.hashCode"))
   override def hashCode(): Int = juuid.hashCode
 
+  /**
+   * Returns a {@code String} object representing this {@code UUID}.
```

Comment:
We probably want links here too instead of `{@code }` blocks, am I right?

Reply:
Absolutely